proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,1028 @@
+
+Collection namespacing is also necessary for scalable endpoint support
+in [LEDS](https://github.com/envoyproxy/envoy/issues/10373).
+
+In the proposal below, we introduce an explicit notion of collections and
+namespaces, expressed through xDS resource name URIs.
+
+#### Per-resource type transport naming semantics
+
+While most xDS resource types permit opaque resource naming at the discretion of
+the control plane, some xDS resources, namely VHDS, have specific resource
+name structure. VHDS requires a specific form
+[form](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/vhds.html?highlight=vhds#virtual-host-resource-naming-convention)
+for referenced resources, `<route configuration name>/<host entry>`. This permits the management
+server to lookup a  missing virtual host with the host entry inside the
+specified `RouteConfiguration` resource name. In this scheme, the route
+configuration name is acting as a namespace for the virtual host name.
+
+It’s likely other resource types in the future will require some structure, e.g.
+[LEDS](https://github.com/envoyproxy/envoy/issues/10373).  Rather than living
+with an arbitrary mix of opaque and structured forms, the URI-based xDS resource
+naming scheme below provides a structured scheme that can serve the
+expression needs of resource types such as VHDS and LEDS.
+
+#### Aliasing
+
+VHDS introduced aliasing to the xDS transport, since the same virtual host
+resource may be known by multiple names, e.g. `routeConfigA/foo.com,
+routeConfigA/bar.com`. This mechanism is not widely used yet elsewhere in xDS
+and it is another example of mixing the transport and data model resource layers
+in xDS.
+
+We introduce a redirection mechanism below that allows a control plane to
+provide aliasing semantics without requiring explicit transport-level support
+for aliasing. When querying for either `routeConfigA/foo.com` or `routeConfigA/bar.com`, the
+management server might issue a redirect causing the xDS client to fetch
+`someCanonicalVirtualHost` resource.
+
+This same mechanism is also useful when supporting resource delegation during
+federation and removes special case treatment of aliasing from the xDS transport
+protocol.
+
+### Related Proposals:
+
+* The original Google Docs design documents:
+  [Full](https://docs.google.com/document/d/1zZav-IYxMO0mP2A7y5XHBa9v0eXyirw1CxrLLSUqWR8/edit#heading=h.w8hw2zbv3jwl)
+  [Summary](https://docs.google.com/document/d/1m5_Q9LUlzvDdImP0jqh1lSTrLMldsApTQX6ibbd3i7c/edit?ts=5ec2fd8a#heading=h.w8hw2zbv3jwl)
+* [xDS transport protocol.](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol)
+
+## Proposal
+
+The proposal below is fully backwards compatible with the existing xDS transport
+protocol and does not require any changes to clients or servers that do not wish
+to opt into the new naming scheme.
+
+### URI-based xDS resource names
+
+The following URI schema is proposed for xDS resource names:
```

Comment:
Might be a good idea to explicitly say that this is an RFC-3986-compliant schema, with a link to https://tools.ietf.org/html/rfc3986.